File: 프로그래머스/lv2/12973. 짝지어 제거하기/짝지어 제거하기.js
function solution(s) {
  const stack = [];

  for (const item of s) {
    // 스택이 비어있으면 아이템 추가
    if (!stack.length) {
      stack.push(item);
    } else {
      if (stack[stack.length - 1] === item) {
        stack.pop();
      } else {
        stack.push(item);
      }
    }
  }

  if (!stack.length) return 1;
  return 0;
}

File: 프로그래머스/lv2/12973. 짝지어 제거하기/짝지어 제거하기.py
def solution(s):
    stack = []
    
    for i in range(len(s)):
        if not len(stack):
            stack.append(s[i])
        else:
            if stack[-1] == s[i]:
                stack.pop()
            else:
                stack.append(s[i])

    if len(stack):
        return 0
    return 1
